torch/csrc/jit/frontend/sugared_value.cpp

```diff
@@ -284,6 +290,16 @@ void SimpleValue::setAttr(
             << "Initialize the field at the top level first";
       }
     } else {
+      // Check and see if it's a setter attribute.
+      auto prop = classType->getProperty(field);
+      if (prop && prop->setter) {
+        auto& g = *m.graph();
```

Comment:
Could you refactor this to use `MethodValue(value_, getter_name)->call(...)` ? If you look at the function there is additional logic there like checking for recursive calls, and setting the source range location that is not being done here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/frontend/sugared_value.h#L448

Reply:
Done.